File: ruby/high-scores/high_scores.rb
class HighScores
  attr_reader :scores

  def initialize(arr)
    @scores = arr
  end

  def latest
    @scores.last
  end

  def highest
    @scores.max
  end

  def top
    @scores.max(3)
  end

  def report
    str = "Your latest score was #{latest}."

    diff = highest - latest

    if diff.positive?
      str << " That's #{diff} short of your personal best!"
    else
      str << " That's your personal best!"
    end

    return str
  end
end

File: javascript/high-scores/high-scores.js
export class HighScores {
    constructor(arr) {
        this.scores = arr.slice();
    }

    get latest() {
        return this.scores[this.scores.length - 1];
    }

    get personalBest() {
        return Math.max(...this.scores);
    }

    get personalTopThree() {
        let sorted = this.scores.slice().sort((a, b) => b - a);
        return sorted.slice(0, 3);
    }
}
